deepfence_worker/cronjobs/neo4j.go

```diff
@@ -505,27 +494,45 @@ func LinkCloudResources(ctx context.Context, task *asynq.Task) error {
 		SET n.linked = true
 		WITH n, apoc.convert.fromJsonList(n.vpc_security_group_ids) as sec_group_ids
 		UNWIND sec_group_ids as subgroup
-		OPTIONAL MATCH (m:CloudResource{id: "aws_vpc_security_group_rule", group_id: subgroup})
-		MATCH (o:Node {node_id:'out-the-internet'})
-		MATCH (p:Node {node_id:'in-the-internet'})
-		WITH m, n, o, p, CASE WHEN m IS NOT NULL THEN [1] ELSE [] END AS make_cat
-		FOREACH (i IN make_cat |
-			MERGE (m) -[:SECURED]-> (n)
-		)
-		WITH m, n, o, p, CASE WHEN m.is_egress AND m.cidr_ipv4 = "0.0.0.0/0" THEN [1] ELSE [] END AS make_cat2
-		FOREACH (i IN make_cat2 |
-			MERGE (n) <-[:PUBLIC]- (o)
-		)
-		WITH m, n, o, p, CASE WHEN NOT m.is_egress AND m.cidr_ipv4 = "0.0.0.0/0" THEN [1] ELSE [] END AS make_cat3
-		FOREACH (i IN make_cat3 |
-			MERGE (n) <-[:PUBLIC]- (p)
-		)`,
+		MATCH (m:CloudResource{id: "aws_vpc_security_group_rule"})
+		WHERE m.node_id ENDS WITH subgroup
```

Comment:
Any reason to check the suffix? Will this not make the query slower?

Reply:
Since `node_id` is indexed, this operation is 2 times faster than doing the previous operation